curp/src/server/raw_curp/mod.rs

```diff
@@ -688,21 +692,22 @@ impl<C: 'static + Command> RawCurp<C> {
         let log_r = self.log.read();
         if next_index <= log_r.base_index {
             // the log has already been compacted
-            let entry = log_r.get(log_r.last_applied).unwrap_or_else(|| {
+            let entry = log_r.get(log_r.last_exe).unwrap_or_else(|| {
                 unreachable!(
                     "log entry {} should not have been compacted yet, needed for snapshot",
-                    log_r.last_applied
+                    log_r.last_as
                 )
             });
+            // TODO: buffer a local snapshot, don't take snapshot every time
```

Comment:
Why do we need to buffer a local snapshot? What's meaning of "local" here?

Reply:
More comments added.